tests/src/App/ElementMap.cpp

```diff
@@ -197,4 +200,129 @@ TEST_F(ElementMapTest, mimicSimpleUnion)
     // "Face3" is the localized name
 }
 
+TEST_F(ElementMapTest, mimicOperationAgainstSelf)
```

Comment:
To be totally honest I don't quite understand the logic of the toponaming algorithm, so I can't really comment if this test is right or wrong.